.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,84 @@
+name: npm-publish
+on:
+  push:
+    branches:
+      - mobx
+
+jobs:
+  publish:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Check version changes
+        uses: EndBug/version-check@v1
+        id: check
+        with:
+          diff-search: true
+
+      - name: Version update detected
+        if: steps.check.outputs.changed == 'true'
+        run: 'echo "Version change found! New version: ${{ steps.check.outputs.version }} (${{ steps.check.outputs.type }})"'
+
+      - name: Set up Node.js for NPM
+        if: steps.check.outputs.changed == 'true'
+        uses: actions/setup-node@v1
+        with:
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        if: steps.check.outputs.changed == 'true'
+        run: npm install
+
+      - name: Lint and build terriajs
+        if: steps.check.outputs.changed == 'true'
+        run: npx gulp lint release
+        env:
+          NODE_OPTIONS: '--max_old_space_size=8192'
+
+      - name: Publish package to NPM
+        if: steps.check.outputs.changed == 'true'
+        run: npm publish --tag mobx
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Tag new version
+        if: steps.check.outputs.changed == 'true'
+        uses: steve9164/annotated-tag@v1
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          tag-name: ${{ steps.check.outputs.version }}
+          tag-message: ${{ steps.check.outputs.version }}
+          tagger-name: Terria Bot via GitHub Actions
+          tagger-email: TerriaBot@users.noreply.github.com
+
+      # These slack success and failure message actions can be combined but it means there'll be no notification on a failed
+      #  version-check (which is probably unlikely to fail). It also allows for a bit better message customisation
+      - name: action-slack-success
```

Comment:
Why is it called `action-slack-success` rather than `Send success notification on Slack` or something? Does the name have special meaning for this one?

Reply:
Very good point. At some point I think I got confused between `id` and `name`.